pystac/utils.py

```diff
@@ -329,14 +329,48 @@ def datetime_to_str(dt: datetime, timespec: str = "auto") -> str:
 def str_to_datetime(s: str) -> datetime:
     """Converts a string timestamp to a :class:`datetime.datetime` instance using
     :meth:`dateutil.parser.parse` under the hood. The input string may be in any
-    format :std:doc:`supported by the parser <parser>`.
+    format :std:doc:`supported by the parser <parser>`. This includes many formats
+    including ISO 8601 and RFC 3339.
 
     Args:
         s (str) : The string to convert to :class:`datetime.datetime`.
+
+    Returns:
+        str: The :class:`datetime.datetime` represented the by the string.
+    """
+    return dateutil.parser.isoparse(s)
+
+
+def iso8601_str_to_datetime(s: str) -> datetime:
+    """Converts a string conforming to ISO 8601 to a :class:`datetime.datetime`
+    instance using :meth:`dateutil.parser.isoparse` under the hood.  STAC requires
+    datetime values to be in RFC 3339 (a stricter subset of ISO 8601) and be represented
+    in the Coordinated Universal Time (UTC) timezone using 'Z' or '+00:00'. This method
+    is more lenient than STAC requires as not to cause errors with string values that
+    are non-conformant.
+
+    Args:
+        s (str) : The string to convert to :class:`datetime.datetime`.
+
+    Returns:
+        datetime: The datetime represented by the ISO8601 (RFC 3339) formatted string.
+
+    Raises:
+        ValueError: If the string is not a valid RFC 3339 string.
     """
     return dateutil.parser.isoparse(s)
```

Comment:
Without ciso8601, these two functions are the same. I'd suggest removing `iso8601_to_datetime`.